### ISSUE TITLE
Misleading flax.linen.scan example

### ISSUE THREAD

**Issue by n2cholas** [google/flax#1294]
Below is the example provided for [`flax.linen.scan`](https://flax.readthedocs.io/en/latest/_autosummary/flax.linen.scan.html#flax.linen.scan):

```python
class SimpleScan(nn.Module):
  @nn.compact
  def __call__(self, c, xs):
    LSTM = nn.scan(nn.LSTMCell,
                   variable_broadcast="params",
                   split_rngs={"params": False})
    return LSTM()(c, xs)

xs = random.uniform(rng_1, (batch_size, features))
carry_0 = nn.LSTMCell.initialize_carry(
    random.PRNGKey(0), (batch_size,), features)
model = SimpleScan()
variables = model.init(key_2, carry_0, xs)
out_state, out_val = model.apply(variables, carry_0, xs)
```
The default `in_axes` for `nn.scan` is `0`, so it seems like this example is scanning over the batch dimension instead of over a sequence.

I believe `xs` needs a sequence dimension in axis 0, as below:

```python
import flax
import flax.linen as nn
from jax import random

class SimpleScan(nn.Module):
  @nn.compact
  def __call__(self, c, xs):
    LSTM = nn.scan(nn.LSTMCell,
                   variable_broadcast="params",
                   split_rngs={"params": False})
    return LSTM()(c, xs)

key_1, key_2, key_3 = random.split(random.PRNGKey(0), 3)
seq_len, batch_size, features = 7, 11, 13

xs = random.uniform(key_1, (seq_len, batch_size, features))
carry_0 = nn.LSTMCell.initialize_carry(key_2, (batch_size,), features)
model = SimpleScan()
variables = model.init(key_3, carry_0, xs)
out_state, out_val = model.apply(variables, carry_0, xs)
```

Is this correct? I've read https://github.com/google/flax/discussions/1283 to verify. Thanks in advance!

**Comment by jheek** [comment]
Yes that looks good. Want to make a PR?

Nit: it's customary to put the batch dim first and putting the sequence length second demonstrates that we can scan over non-leading axes :)

**Comment by n2cholas** [comment]
Will do!
